scipy-stubs/sparse/_construct.pyi

```diff
@@ -38,15 +40,16 @@ __all__ = [
     "vstack",
 ]
 
+_Numeric: TypeAlias = npc.number | np.bool_
+_AnyFloat64DTypeNotNone: TypeAlias = type[op.JustFloat] | onp.ToDType[npc.floating64] | f8_code
+
 _T = TypeVar("_T")
 _SCT = TypeVar("_SCT", bound=_Numeric, default=Any)
 _SCT0 = TypeVar("_SCT0", bound=_Numeric)
 _ShapeT = TypeVar("_ShapeT", bound=tuple[int, *tuple[int, ...]], default=tuple[Any, ...])
 
-_Numeric: TypeAlias = npc.number | np.bool_
-
-_ToArray1D: TypeAlias = Seq[_SCT] | onp.CanArrayND[_SCT]
-_ToArray2D: TypeAlias = Seq[Seq[_SCT | int] | onp.CanArrayND[_SCT]] | onp.CanArrayND[_SCT]
+_ToArray1D: TypeAlias = onp.ToArray1D[complex, _SCT]
+_ToArray2D: TypeAlias = onp.ToArray2D[complex, _SCT]
```

Comment:
This way, `_ToArray1D[np.bool_]` will accept `[1j, 2j]`, is that intentional? Because if not, `onp.CanArray{1,2}D` is probably more appropriate.